lib/index.ts

```diff
@@ -246,6 +265,91 @@ class ProgressReporter {
 		};
 	}
 
+	buildProgress(onProgress: ProgressCallback): ProgressCallback {
+		const progressRenderer = this.renderProgress;
+		let lastStep = 1;
+		let step = 0;
+		let total = 1;
+		let lastPercentage = 0;
+
+		let progressCallback: ProgressCallback = () => void 0;
+
+		return (evt: any) => {
+			let stream;
+			let id;
+			let status;
+			try {
+				({ id, stream, status } = evt);
+				if (id == null) {
+					id = '';
+				}
+				if (status == null) {
+					status = '';
+				}
+				if (stream == null) {
+					stream = '';
+				}
+
+				// Check for a `Step n/total` message on the stream
+				const streamComponents = stream.match(/^Step (\d+)\/(\d+)\s*:\s*(.+)/);
+				if (streamComponents != null) {
+					let instruction = '';
+					[, step, total, instruction] = streamComponents;
+
+					// Normalize the lastPercentage by the number of steps
+					lastPercentage = Math.floor((100.0 * (step - 1)) / total);
```

Comment:
Out of interest, does it make a difference to write `100` or `100.0`? I know it can make a difference in Python where `100.0` would be a literal `float` and `100` would be a literal `int`, but what about JavaScript? -
```
$ node
> typeof(100)
'number'
> typeof(100.0)
'number'
```

Reply:
I'm just used to it from programming on other languages, where `/` is integer division unless one of the operands is a float. Actually I didn't know it was not necessary in javascript.